core/src/apps/cardano/auxiliary_data.py

```diff
@@ -0,0 +1,255 @@
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.messages import (
+    CardanoAddressType,
+    CardanoAuxiliaryDataType,
+    CardanoMetadataType,
+)
+
+from apps.common import cbor
+
+from ..common.seed import remove_ed25519_prefix
+from .address import derive_address_bytes, derive_human_readable_address
+from .helpers import INVALID_AUXILIARY_DATA, bech32
+from .helpers.bech32 import HRP_PUBLIC_KEY
+from .helpers.paths import SCHEMA_STAKING_ANY_ACCOUNT
+from .layout import confirm_catalyst_registration, show_auxiliary_data_hash
+
+if False:
+    from typing import Dict, List, Optional, Tuple, Union
+    from trezor import wire
+
+    from trezor.messages.CardanoCatalystRegistrationParametersType import (
+        CardanoCatalystRegistrationParametersType,
+    )
+    from trezor.messages.CardanoTxAuxiliaryDataType import CardanoTxAuxiliaryDataType
+
+    CatalystRegistrationBody = Dict[int, Union[bytes, int]]
```

Comment:
I'd name this `CatalystRegistrationPayload` and the outer structure simply `CatalystRegistration` . I'd generally "abstain" from baking the structure type into the variable name unless it's well justified (e.g. completely missing typing or that it's  giving some crucial clarification in the code), as the main purpose of a variable name is IMHO to give the underlying structure a high level meaning for the reader and the structure type is kind of a low-level detail, normally not that relevant to understand the code which is IMHO the case here as well

Reply:
Nice, cleaner. Done in https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/02a3b6f62c4f15bba6d5ed2f6ef0be7c19ee17e6